lnrpc/walletrpc/walletkit_server.go

```diff
@@ -270,6 +271,12 @@ type WalletKit struct {
 	UnimplementedWalletKitServer
 
 	cfg *Config
+
+	// As we allow rpc requests into the server before InjectDependencies
+	// has been executed, the read lock should be held when accessing values
+	// from the cfg.
+	// The write lock should be held when setting the cfg.
+	sync.RWMutex
```

Comment:
it's unclear to me why this is lock is needed. If we are populating the cfg even earlier now, then shouldnt it be safe as it was before? why would we allow requests before InjectDependencies?

Reply:
My motivation behind adding that was the following:
> why would we allow requests before InjectDependencies? 

You're right that we do not allow requests to most functions here before `InjectDependencies` have been called the second time to finalize the dependencies. However we do allow calls to the `SignCoordinatorStreams` RPC, and since that function accesses the `cfg` we want to ensure that we avoid a race and that the `cfg` isn't read at the exact time that it's being written to in `InjectDependencies`. As the `SignCoordinatorStreams` can be called multiple times (even after the remote signer has connected), that could happen if we don't hold a lock when reading the `cfg` in `SignCoordinatorStreams` and when writting to it in `InjectDependencies`.

I didn't want to create logic that required such specific knowledge just for that, so I opted to what I thought was an easier rule to understand, i.e. hold the lock when reading the `cfg` and when writting to it, which therefore also got applied to all other RPC functions here.

My goal was to try to avoid to leak knowledge about the exact startup flow into this file, i.e. that we only call `InjectDependencies` twice during the startup flow for this server, and that it's only `SignCoordinatorStreams` that's callable between the first and second call since the rpc interceptor will block any other calls. My motivation was that the more knowledge about the flow outside of this file it requires, the harder the code becomes to read and reason about in this file.

-----

Though given your comment, that choice obviously creates confusion the other way around 😅. 

So I'd be ok with dropping the locking of this `RWMutex` in the other functions `SignCoordinatorStreams` if you think that's better :). I've pushed a fixup commit for that, so let me know what you prefer :).